### PR TITLE
Draft info about mapfirst process, how to test

### PR DESCRIPTION
**Pull request by robander**
Signed-off-by: Robert D Anderson <robander@us.ibm.com>

Rewrite of the topic on map-first processing introduced in 2.5, to provide more detail, information about how to test in your own custom transformation types, as well as known limitations.

### CODE REVIEW

**Comment by robander** [comment]
> the notion of colleagues as internal "customers" is less common outside of large organizations

Yes, of course. Slightly different use case, exactly the same procedure + result, agree we should take out the "customers" bit.

**Comment by infotexture** [comment]
Rendered output is available on the staging server:

http://infotexture.github.io/dita-ot.github.io/dev/dev_ref/map-first-preprocessing.html

File: dev_ref/map-first-preprocessing.dita [hunk 1]
```diff
@@ -2,33 +2,114 @@
...
+                   html5.map,
+                   html5.css"/>
+</codeblock>
+</section>
+<section>
+<title>Known limitations</title>
+<ol>
+<li>The <codeph>preprocess2</codeph> implementation details are subject to change; dependencies
+within <codeph>preprocess2</codeph> may be renamed or removed based on feedback.</li>
+<li>Subject schemes and delayed key resolution have not yet been implemented, but support for at
```
Comment (jelovirt) [inline dev_ref/map-first-preprocessing.dita]: Subject schemes are supported, dita-ot/dita-ot#2626

File: dev_ref/map-first-preprocessing.dita [hunk 2]
```diff
@@ -2,33 +2,114 @@
...
+least subject schemes will likely be added in an upcoming release.</li>
+<li>The <codeph>preprocess2</codeph> implementation does not currently support maps that use the
```
Comment (jelovirt) [inline dev_ref/map-first-preprocessing.dita]: Fixed in dita-ot/dita-ot@69bfcd3cd0779d21938ce9ba31ee9f2361a3ba4c

**Pull request merged**